CMGTools/TTHAnalysis/python/tools/angular_vars.py

```diff
@@ -0,0 +1,121 @@
+from CMGTools.TTHAnalysis.treeReAnalyzer import *
+
+class angular_vars:
+    def __init__(self):
+ 	self.branches = [ "L_phi", 
+			  "B_loose_phi", "B_medium_phi", "B_tight_phi", 
+			  "DR_l_b_loose1", "DR_l_b_medium1", "DR_l_b_tight1", 
+			  "DR_l_b_loose2", "DR_l_b_medium2", "DR_l_b_tight2",
+			  "pTl_DR_l_b_loose1", "pTl_DR_l_b_medium1", "pTl_DR_l_b_tight1", 
+			  "pTl_DR_l_b_loose2", "pTl_DR_l_b_medium2", "pTl_DR_l_b_tight2",  
+			  "L_pt" ]
+    def listBranches(self):
+        return self.branches[:]	
+    def __call__(self,event):
+        # make python lists as Collection does not support indexing in slices
+	leps = [l for l in Collection(event,"LepGood","nLepGood",4)]
+        jets = [j for j in Collection(event,"Jet","nJet25",8)]
+	bloose  = [j for j in jets if j.btagCSV > 0.605]
+        bmedium = [j for j in jets if j.btagCSV > 0.890]
+	btight = [j for j in jets if j.btagCSV > 0.970]
+	(met, metphi)  = event.met_pt, event.met_phi
+	njet = len(jets); nlep = len(leps); nbloose = len(bloose); nbmedium = len(bmedium); nbtight = len(btight);
+	ret = dict([(name,0.0) for name in self.branches])
+	
+	#L_phi:
+	
+	l_phi = 0
+        for i in range(len(leps)):
+	    if leps[i].relIso03 < 0.1:
+	       l_phi += leps[i].pt * abs(deltaPhi(metphi,leps[i].phi))
+	ret["L_phi"] = l_phi
+	
+		
+	#B_phi:
+	
+	b_loose_phi = 0
+	for j in range(len(bloose)):
+	    b_loose_phi += bloose[j].pt * abs(deltaPhi(metphi,bloose[j].phi))
+	ret["B_loose_phi"] = b_loose_phi
+	b_medium_phi = 0
+	for k in range(len(bmedium)):
+	    b_medium_phi += bmedium[k].pt * abs(deltaPhi(metphi,bmedium[k].phi))
+	ret["B_medium_phi"] = b_medium_phi
+	b_tight_phi = 0
+	for l in range(len(btight)):
+	    b_tight_phi += btight[l].pt * abs(deltaPhi(metphi,btight[l].phi))
+	ret["B_tight_phi"] = b_tight_phi
+	
+	#DR_l_b_1:
+		
+	if nbloose >= 1:
+	   if leps[0].relIso03 < 0.1:
+              ret["DR_l_b_loose1"] = min([deltaR(m,leps[0]) for m in bloose]) if nlep >= 1 else 0;
+	if nbmedium >= 1:
+	   if leps[0].relIso03 < 0.1:
+              ret["DR_l_b_medium1"] = min([deltaR(n,leps[0]) for n in bmedium]) if nlep >= 1 else 0;
+	if nbtight >= 1:
+	   if leps[0].relIso03 < 0.1:
+              ret["DR_l_b_tight1"] = min([deltaR(p,leps[0]) for p in btight]) if nlep >= 1 else 0;
+	      
+	#DR_l_b_2:
+
+	if nbloose >= 1:
+	   if leps[1].relIso03 < 0.1:
+              ret["DR_l_b_loose2"] = min([deltaR(q,leps[1]) for q in bloose]) if nlep >= 1 else 0;
```

Comment:
this and below should be  `if nlep >= 2` rather than `if nlep >= 1` since you access leps[1]